tensorboard/plugins/mesh_visualizer/mesh_plugin.py

```diff
@@ -0,0 +1,253 @@
+    all_runs = self._multiplexer.PluginRunToTagToContent(MeshPlugin.plugin_name)
+
+    # tagToContent is itself a dictionary mapping tag name to string
+    # SummaryMetadata.plugin_data.content. Retrieve the keys of that dictionary
+    # to obtain a list of tags associated with each run. For each tag, estimate
+    # the number of samples.
+    self._tag_to_instance_tags = collections.defaultdict(list)
+    self._instance_tag_to_metadata = dict()
+    for _, tag_to_content in six.iteritems(all_runs):
+      for tag, content in six.iteritems(tag_to_content):
+        meta = metadata.parse_plugin_metadata(content)
+        self._instance_tag_to_metadata[tag] = meta
+        # Remember instance_name (instance_tag) for future reference.
+        self._tag_to_instance_tags[meta.name].append(tag)
+        self._instance_tag_to_tag[tag] = meta.name
+
+  @wrappers.Request.application
+  def _serve_tags(self, request):
+    """A route (HTTP handler) that returns a response with tags.
+
+    Args:
+      request: The werkzeug.Request object.
+
+    Returns:
+      A response that contains a JSON object. The keys of the object
+      are all the runs. Each run is mapped to a (potentially empty)
+      list of all tags that are relevant to this plugin.
+    """
+    # This is a dictionary mapping from run to (tag to string content).
+    # To be clear, the values of the dictionary are dictionaries.
+    all_runs = self._multiplexer.PluginRunToTagToContent(
+        MeshPlugin.plugin_name)
+
+    # Make sure we populate tags mapping structures.
+    self.prepare_metadata()
+
+    # tagToContent is itself a dictionary mapping tag name to string
+    # SummaryMetadata.plugin_data.content. Retrieve the keys of that dictionary
+    # to obtain a list of tags associated with each run. For each tag estimate
+    # number of samples.
+    response = dict()
+    for run, tag_to_content in six.iteritems(all_runs):
+      response[run] = dict()
+      for instance_tag, _ in six.iteritems(tag_to_content):
+        # Make sure we only operate on user-defined tags here.
+        tag = self._instance_tag_to_tag[instance_tag]
+        meta = self._instance_tag_to_metadata[instance_tag]
+        # Shape should be at least BxNx3 where B represents the batch dimensions
+        # and N - the number of points, each with x,y,z coordinates.
+        assert len(meta.shape) >= 3
+        response[run][tag] = {'samples': meta.shape[0]}
+    return http_util.Respond(request, response, 'application/json')
+
+  def get_plugin_apps(self):
+    """Gets all routes offered by the plugin.
+
+    This method is called by TensorBoard when retrieving all the
+    routes offered by the plugin.
+    Returns:
+      A dictionary mapping URL path to route that handles it.
+    """
+    # Note that the methods handling routes are decorated with
+    # @wrappers.Request.application.
+    return {
+        '/tags': self._serve_tags,
```

Comment:
Can you please provide http_api.md that details the contract between backend and the frontend?
Here is an example: https://github.com/tensorflow/tensorboard/blob/1b623e7c914558b0e521d946fccee2b8ebdb53b9/tensorboard/plugins/scalar/http_api.md

It will help us understand the code better.